File: First Unique Character in a String/Suggested.java
class Solution {

    public static int firstUniqChar(String s) {
        int[] counter = new int[26];
        for (int i=0; i<s.length(); i++) {
            int index = s.charAt(i) - 'a';
            counter[index]++;
        }
        for (int i=0; i<s.length(); i++) {
            int index = s.charAt(i) - 'a';
            if (counter[index] == 1) {
                return i;
            }
        }
        return -1;
    }
}

File: First Unique Character in a String/suggested.py
class Solution:
    def firstUniqChar(self, str):
        counter={}
        for char in str:
            counter[char] = counter.get(char,0)+1
        for i in range(len(str)):
            if counter[str[i]] == 1:
                return i
        return -1


if __name__ == "__main__":
    a = Solution()
    print(a.firstUniqChar("leetcode"))
    print(a.firstUniqChar("loveleetcode"))
    print(a.firstUniqChar("aabb"))
